### PR TITLE
Synchronous improvements

### PR DESCRIPTION
**Pull request by gossen**
1. Change synchronous systems representation in Backend
2. Preserve clocks hierarchy
3. Fix removed equations partitioning
4. Add  interval and boolean clocks support in compiler and C simulation runtime

### CODE REVIEW

**Comment by adrpo** [comment]
@lochel, @bernhard-thiele can you review this pull request so we can merge it in if is fine?

**Comment by bernhard-thiele** [comment]
I had a quick look on it. I haven't worked at the back-end part of the synchronous features, so I don't know the code there and I can not really review it. I can only say that I didn't notice anything that appeared  obviously unreasonably to me.

**Comment by lochel** [comment]
After my holidays (end of august) I will check the back end changes.

**Comment by lochel** [comment]
I did a rough review of the changes and I think they are good to push.

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2015-08-31_10-29-13](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/663/console).

File: Compiler/SimCode/SimCodeUtil.mo [hunk 1]
```diff
@@ -5980,6 +5979,9 @@ algorithm
   BackendDAE.DAE(eqs=systs1, shared=BackendDAE.SHARED(knownVars=knvars1, externalObjects=extvars1, aliasVars=aliasVars1)) := inSimDAE;
   BackendDAE.DAE(eqs=systs2, shared=BackendDAE.SHARED(knownVars=knvars2, externalObjects=extvars2, aliasVars=aliasVars2)) := inInitDAE;
 
+  systs1 := List.filterOnFalse(systs1, BackendDAEUtil.isClockedSyst);
+  systs2 := List.filterOnFalse(systs1, BackendDAEUtil.isClockedSyst);
```
Comment (lochel) [inline Compiler/SimCode/SimCodeUtil.mo]: Oops – I overlooked this copy&paste fault.

**Pull request closed without merging**